packages/client/hmi-client/src/services/project.ts

```diff
@@ -275,6 +275,43 @@ function getAssetIcon(type: AssetType | string | null): string | Component {
 	return 'circle';
 }
 
+/**
+ * Find projects by KNN (k-nearest neighbors) search
+ *
+ * @param {string} query - the search query
+ * @param {number} k - the number of neighbors to search for
+ * @param {number} page - the page number
+ * @param {number} pageSize - the number of projects to return per page
+ * @param {number} limit - the number of projects to return
+ * @returns {Array<Project>} - the list of projects
+ */
+async function findProjects(
+	query: string,
+	k?: number,
+	page?: number,
+	pageSize?: number,
+	limit?: number
+): Promise<ProjectSearchResponse[]> {
```

Comment:
This kind of optional interface is misleading, because you even though "k" is marked as optional, you can't skip "k" if you want to specify "limits".

I would recommend an option object
```
findProject(query, {
  k?
  page?
  pageSIze?
  limit?
}) = ...
```